#VRML_SIM R2023b utf8
# license: Apache License 2.0
# license url: http://www.apache.org/licenses/LICENSE-2.0
# This is a proto file for Webots for the 4WheelConfig
# Extracted from: /home/pratik/ros_ws/colcon_ws_AMR/src/my_project/protos/4_wheel_config.urdf

PROTO FourWheelConfig [
  field  SFVec3f     translation     0 0 -0.25
  field  SFRotation  rotation        0 0 1 0
  field  SFString    name            "FourWheelConfig"  # Is `Robot.name`.
  field  SFString    controller      "obstacle_avoidance"          # Is `Robot.controller`.
  field  MFString    controllerArgs  []              # Is `Robot.controllerArgs`.
  field  SFString    customData      ""              # Is `Robot.customData`.
  field  SFBool      supervisor      FALSE           # Is `Robot.supervisor`.
  field  SFBool      synchronization TRUE            # Is `Robot.synchronization`.
  field  SFBool      selfCollision   FALSE           # Is `Robot.selfCollision`.
  field  MFNode      extensions        []              # Extend the robot with new nodes at the end of the arm.\
  
]
{
  Robot {
    translation IS translation
    rotation IS rotation
    controller IS controller
    controllerArgs IS controllerArgs
    customData IS customData
    supervisor IS supervisor
    synchronization IS synchronization
    selfCollision IS selfCollision
    
    children [
      Solid {
        translation 0.233000 0.116500 0.050000
        rotation 0 0 0 0
        children [
          DEF wheeled_brick_housing_visual CadShape {
            url "/home/pratik/ros_ws/colcon_ws_AMR/src/robile_description/meshes/robile_brick_housing/wheeled_brick_housing.dae"
          }
          DEF lid_visual CadShape {
            url "/home/pratik/ros_ws/colcon_ws_AMR/src/robile_description/meshes/robile_brick_housing/lid.dae"
          }
          DEF lid_cover_red_visual CadShape {
            url "/home/pratik/ros_ws/colcon_ws_AMR/src/robile_description/meshes/robile_brick_housing/lid_cover_red.dae"
          }
          HingeJoint {
            jointParameters HingeJointParameters {
              axis 0.000000 0.000000 1.000000
            }
            device [
              RotationalMotor {
                name "robile_1_drive_pivot_joint"
                maxTorque 10000
              }
              PositionSensor {
                name "robile_1_drive_pivot_joint_sensor"
              }
            ]
            endPoint Solid {
              children [
                DEF kelo_drive_base_visual CadShape {
                  url "/home/pratik/ros_ws/colcon_ws_AMR/src/robile_description/meshes/kelo_drive/kelo_drive_base.dae"
                }
                HingeJoint {
                  jointParameters HingeJointParameters {
                    axis 0.000000 1.000000 0.000000
                    anchor -0.010000 0.039326 0.001514
                  }
                  device [
                    RotationalMotor {
                      name "robile_1_drive_left_hub_wheel_joint"
                      maxTorque 0
                    }
                    PositionSensor {
                      name "robile_1_drive_left_hub_wheel_joint_sensor"
                    }
                  ]
                  endPoint Solid {
                    translation -0.010000 0.039326 0.001514
                    rotation 0 0 0 0
                    children [
                      DEF kelo_drive_wheel_visual CadShape {
                        url "/home/pratik/ros_ws/colcon_ws_AMR/src/robile_description/meshes/kelo_drive/kelo_drive_wheel.dae"
                      }
                    ]
                    name "robile_1_drive_left_hub_wheel_link"
                      boundingObject Pose {
                      translation 0 0 0
                      rotation  1 0 0 1.57
                      children [
                          Cylinder {
                      radius 0.0515
                      height 0.035
                          }
                      ]
                    }

                    physics Physics {
                      density -1
                      mass 10.000000
                      centerOfMass [ 0.000000 0.000000 0.000000 ]
                      inertiaMatrix [
                        9.048167e-02 9.048167e-02 9.048167e-02
                        0.000000e+00 0.000000e+00 0.000000e+00
                      ]
                    }
                  }
                }
                HingeJoint {
                  jointParameters HingeJointParameters {
                    axis 0.000000 1.000000 0.000000
                    anchor -0.010000 -0.039326 0.001514
                  }
                  device [
                    RotationalMotor {
                      name "robile_1_drive_right_hub_wheel_joint"
                      maxTorque 0
                    }
                    PositionSensor {
                      name "robile_1_drive_right_hub_wheel_joint_sensor"
                    }
                  ]
                  endPoint Solid {
                    translation -0.010000 -0.039326 0.001514
                    rotation 0 0 0 0
                    children [
                      USE kelo_drive_wheel_visual
                    ]
                    name "robile_1_drive_right_hub_wheel_link"
                    
                      boundingObject Pose {
                      translation 0 0 0
                      rotation  1 0 0 1.57
                      children [
                          Cylinder {
                      radius 0.0515
                      height 0.035
                          }
                      ]
                    }
                    physics Physics {
                      density -1
                      mass 10.000000
                      centerOfMass [ 0.000000 0.000000 0.000000 ]
                      inertiaMatrix [
                        9.048167e-02 9.048167e-02 9.048167e-02
                        0.000000e+00 0.000000e+00 0.000000e+00
                      ]
                    }
                  }
                }
              ]
              name "robile_1_drive_pivot_link"

              physics Physics {
                density -1
                mass 10.000000
                centerOfMass [ 0.000000 0.000000 0.000000 ]
                inertiaMatrix [
                  9.048167e-02 9.048167e-02 9.048167e-02
                  0.000000e+00 0.000000e+00 0.000000e+00
                ]
              }
            }
          }
        ]
        name "robile_1_link"

        physics Physics {
          density -1
          mass 4.554000
          centerOfMass [ 0.000050 0.000090 0.120480 ]
          inertiaMatrix [
            5.120000e-02 5.101200e-02 6.274100e-02
            6.200000e-06 2.240000e-05 2.700000e-05
          ]
        }

      }
      Solid {
 
        translation 0.233000 -0.116500 0.050000
        rotation 0 0 0 0
        children [
          USE wheeled_brick_housing_visual
          USE lid_visual
          USE lid_cover_red_visual
          HingeJoint {
            jointParameters HingeJointParameters {
              axis 0.000000 0.000000 1.000000
            }
            device [
              RotationalMotor {
                name "robile_2_drive_pivot_joint"
                maxTorque 0
              }
              PositionSensor {
                name "robile_2_drive_pivot_joint_sensor"
              }
            ]
            endPoint Solid {
              
              children [
                USE kelo_drive_base_visual
                HingeJoint {
                  jointParameters HingeJointParameters {
                    axis 0.000000 1.000000 0.000000
                    anchor -0.010000 0.039326 0.001514
                  }
                  device [
                    RotationalMotor {
                      name "robile_2_drive_left_hub_wheel_joint"
                      maxTorque 0
                    }
                    PositionSensor {
                      name "robile_2_drive_left_hub_wheel_joint_sensor"
                    }
                  ]
                  endPoint Solid {
                    translation -0.010000 0.039326 0.001514
                    rotation 0 0 0 0
                    children [
                      USE kelo_drive_wheel_visual
                    ]
                    name "robile_2_drive_left_hub_wheel_link"
                      boundingObject Pose {
                      translation 0 0 0
                      rotation  1 0 0 1.57
                      children [
                          Cylinder {
                      radius 0.0515
                      height 0.035
                          }
                      ]
                    }
                    physics Physics {
                      density -1
                      mass 10.000000
                      centerOfMass [ 0.000000 0.000000 0.000000 ]
                      inertiaMatrix [
                        9.048167e-02 9.048167e-02 9.048167e-02
                        0.000000e+00 0.000000e+00 0.000000e+00
                      ]
                      
                    }
                  }
                }
                HingeJoint {
                  jointParameters HingeJointParameters {
                    axis 0.000000 1.000000 0.000000
                    anchor -0.010000 -0.039326 0.001514
                  }
                  device [
                    RotationalMotor {
                      name "robile_2_drive_right_hub_wheel_joint"
                      maxTorque 0
                    }
                    PositionSensor {
                      name "robile_2_drive_right_hub_wheel_joint_sensor"
                    }
                  ]
                  endPoint Solid {
                    translation -0.010000 -0.039326 0.001514
                    rotation 0 0 0 0
                    children [
                      USE kelo_drive_wheel_visual
                    ]
                    name "robile_2_drive_right_hub_wheel_link"
                      boundingObject Pose {
                      translation 0 0 0
                      rotation  1 0 0 1.57
                      children [
                          Cylinder {
                      radius 0.0515
                      height 0.035
                          }
                      ]
                    }
                    physics Physics {
                      density -1
                      mass 10.000000
                      centerOfMass [ 0.000000 0.000000 0.000000 ]
                      inertiaMatrix [
                        9.048167e-02 9.048167e-02 9.048167e-02
                        0.000000e+00 0.000000e+00 0.000000e+00
                      ]
                    }
                  }
                }
              ]
              name "robile_2_drive_pivot_link"

              physics Physics {
                density -1
                mass 10.000000
                centerOfMass [ 0.000000 0.000000 0.000000 ]
                inertiaMatrix [
                  9.048167e-02 9.048167e-02 9.048167e-02
                  0.000000e+00 0.000000e+00 0.000000e+00
                ]
              }
            }
          }
        ]
        name "robile_2_link"

        physics Physics {
          density -1
          mass 4.554000
          centerOfMass [ 0.000050 0.000090 0.120480 ]
          inertiaMatrix [
            5.120000e-02 5.101200e-02 6.274100e-02
            6.200000e-06 2.240000e-05 2.700000e-05
          ]
        }
      }
      Solid {
        translation 0.000000 0.116500 0.050000
        rotation 0 0 0 0
        children [
          DEF general_brick_housing_visual CadShape {
            url "/home/pratik/ros_ws/colcon_ws_AMR/src/robile_description/meshes/robile_brick_housing/general_brick_housing.dae"
          }
          USE lid_visual
          DEF lid_cover_yellow_visual CadShape {
            url "/home/pratik/ros_ws/colcon_ws_AMR/src/robile_description/meshes/robile_brick_housing/lid_cover_yellow.dae"
          }
        ]
        name "robile_3_link"

        physics Physics {
          density -1
          mass 4.554000
          centerOfMass [ 0.000050 0.000090 0.120480 ]
          inertiaMatrix [
            5.120000e-02 5.101200e-02 6.274100e-02
            6.200000e-06 2.240000e-05 2.700000e-05
          ]
        }
      }
      Solid {
        translation 0.000000 -0.116500 0.050000
        rotation 0 0 0 0
        children [
          USE general_brick_housing_visual
          USE lid_visual
          DEF lid_cover_green_visual CadShape {
            url "/home/pratik/ros_ws/colcon_ws_AMR/src/robile_description/meshes/robile_brick_housing/lid_cover_green.dae"
          }
        ]
        name "robile_4_link"

        physics Physics {
          density -1
          mass 4.554000
          centerOfMass [ 0.000050 0.000090 0.120480 ]
          inertiaMatrix [
            5.120000e-02 5.101200e-02 6.274100e-02
            6.200000e-06 2.240000e-05 2.700000e-05
          ]
        }

      }
      Solid {
        translation -0.233000 0.116500 0.050000
        rotation 0 0 0 0
        children [
          USE wheeled_brick_housing_visual
          USE lid_visual
          USE lid_cover_red_visual
          HingeJoint {
            jointParameters HingeJointParameters {
              axis 0.000000 0.000000 1.000000
            }
            device [
              RotationalMotor {
                name "robile_5_drive_pivot_joint"
                maxTorque 10000
              }
              PositionSensor {
                name "robile_5_drive_pivot_joint_sensor"
              }
            ]
            endPoint Solid {
              children [
                USE kelo_drive_base_visual
                HingeJoint {
                  jointParameters HingeJointParameters {
                    axis 0.000000 1.000000 0.000000
                    anchor -0.010000 0.039326 0.001514
                  }
                  device [
                    RotationalMotor {
                      name "robile_5_drive_left_hub_wheel_joint"
                      maxTorque 10000
                    }
                    PositionSensor {
                      name "robile_5_drive_left_hub_wheel_joint_sensor"
                    }
                  ]
                  endPoint Solid {
                    translation -0.010000 0.039326 0.001514
                    rotation 0 0 0 0
                    children [
                      USE kelo_drive_wheel_visual
                    ]
                    name "robile_5_drive_left_hub_wheel_link"
                      boundingObject Pose {
                      translation 0 0 0
                      rotation  1 0 0 1.57
                      children [
                          Cylinder {
                      radius 0.0515
                      height 0.035
                          }
                      ]
                    }
                    physics Physics {
                      density -1
                      mass 10.000000
                      centerOfMass [ 0.000000 0.000000 0.000000 ]
                      inertiaMatrix [
                        9.048167e-02 9.048167e-02 9.048167e-02
                        0.000000e+00 0.000000e+00 0.000000e+00
                      ]
                    }
                  }
                }
                HingeJoint {
                  jointParameters HingeJointParameters {
                    axis 0.000000 1.000000 0.000000
                    anchor -0.010000 -0.039326 0.001514
                  }
                  device [
                    RotationalMotor {
                      name "robile_5_drive_right_hub_wheel_joint"
                      maxTorque 10000
                    }
                    PositionSensor {
                      name "robile_5_drive_right_hub_wheel_joint_sensor"
                    }
                  ]
                  endPoint Solid {
                    translation -0.010000 -0.039326 0.001514
                    rotation 0 0 0 0
                    children [
                      USE kelo_drive_wheel_visual
                    ]
                    name "robile_5_drive_right_hub_wheel_link"
                      boundingObject Pose {
                      translation 0 0 0
                      rotation  1 0 0 1.57
                      children [
                          Cylinder {
                      radius 0.0515
                      height 0.035
                          }
                      ]
                    }
                    physics Physics {
                      density -1
                      mass 10.000000
                      centerOfMass [ 0.000000 0.000000 0.000000 ]
                      inertiaMatrix [
                        9.048167e-02 9.048167e-02 9.048167e-02
                        0.000000e+00 0.000000e+00 0.000000e+00
                      ]
                    }
                  }
                }
              ]
              name "robile_5_drive_pivot_link"

              physics Physics {
                density -1
                mass 10.000000
                centerOfMass [ 0.000000 0.000000 0.000000 ]
                inertiaMatrix [
                  9.048167e-02 9.048167e-02 9.048167e-02
                  0.000000e+00 0.000000e+00 0.000000e+00
                ]
              }

            }
          }
        ]
        name "robile_5_link"

        physics Physics {
          density -1
          mass 4.554000
          centerOfMass [ 0.000050 0.000090 0.120480 ]
          inertiaMatrix [
            5.120000e-02 5.101200e-02 6.274100e-02
            6.200000e-06 2.240000e-05 2.700000e-05
          ]
        }

      }
      Solid {
        translation -0.233000 -0.116500 0.050000

        rotation 0 0 0 0
        children [
          USE wheeled_brick_housing_visual
          USE lid_visual
          USE lid_cover_red_visual
          HingeJoint {
            jointParameters HingeJointParameters {
              axis 0.000000 0.000000 1.000000
            }
            device [
              RotationalMotor {
                name "robile_6_drive_pivot_joint"
                maxTorque 10000
              }
              PositionSensor {
                name "robile_6_drive_pivot_joint_sensor"
              }
            ]
            endPoint Solid {
              children [
                USE kelo_drive_base_visual
                HingeJoint {
                  jointParameters HingeJointParameters {
                    axis 0.000000 1.000000 0.000000
                    anchor -0.010000 0.039326 0.001514
                  }
                  device [
                    RotationalMotor {
                      name "robile_6_drive_left_hub_wheel_joint"
                      maxTorque 10000
                    }
                    PositionSensor {
                      name "robile_6_drive_left_hub_wheel_joint_sensor"
                    }
                  ]
                  endPoint Solid {
                    translation -0.010000 0.039326 0.001514
                    rotation 0 0 0 0
                    children [
                      USE kelo_drive_wheel_visual
                    ]
                    name "robile_6_drive_left_hub_wheel_link"
                      boundingObject Pose {
                      translation 0 0 0
                      rotation  1 0 0 1.57
                      children [
                          Cylinder {
                      radius 0.0515
                      height 0.035
                          }
                      ]
                    }
                    physics Physics {
                      density -1
                      mass 10.000000
                      centerOfMass [ 0.000000 0.000000 0.000000 ]
                      inertiaMatrix [
                        9.048167e-02 9.048167e-02 9.048167e-02
                        0.000000e+00 0.000000e+00 0.000000e+00
                      ]
                    }
                  }
                }
                HingeJoint {
                  jointParameters HingeJointParameters {
                    axis 0.000000 1.000000 0.000000
                    anchor -0.010000 -0.039326 0.001514
                  }
                  device [
                    RotationalMotor {
                      name "robile_6_drive_right_hub_wheel_joint"
                      maxTorque 10000
                    }
                    PositionSensor {
                      name "robile_6_drive_right_hub_wheel_joint_sensor"
                    }
                  ]
                  endPoint Solid {
                    translation -0.010000 -0.039326 0.001514
                    rotation 0 0 0 0
                    children [
                      USE kelo_drive_wheel_visual
                    ]
                    name "robile_6_drive_right_hub_wheel_link"
                      boundingObject Pose {
                      translation 0 0 0
                      rotation  1 0 0 1.57
                      children [
                          Cylinder {
                      radius 0.0515
                      height 0.035
                          }
                      ]
                    }
                    physics Physics {
                      density -1
                      mass 10.000000
                      centerOfMass [ 0.000000 0.000000 0.000000 ]
                      inertiaMatrix [
                        9.048167e-02 9.048167e-02 9.048167e-02
                        0.000000e+00 0.000000e+00 0.000000e+00
                      ]
                    }
                  }
                }
              ]
              name "robile_6_drive_pivot_link"

              physics Physics {
                density -1
                mass 10.000000
                centerOfMass [ 0.000000 0.000000 0.000000 ]
                inertiaMatrix [
                  9.048167e-02 9.048167e-02 9.048167e-02
                  0.000000e+00 0.000000e+00 0.000000e+00
                ]
              }
            }
          }
        ]
        name "robile_6_link"

        physics Physics {
          density -1
          mass 4.554000
          centerOfMass [ 0.000050 0.000090 0.120480 ]
          inertiaMatrix [
            5.120000e-02 5.101200e-02 6.274100e-02
            6.200000e-06 2.240000e-05 2.700000e-05
          ]
        }
      }
          Group {
            children IS extensions
          }
    ]
    name IS name
    model "FourWheelConfig"
  boundingObject Pose {
    translation 0 0 0.1
    rotation 0 0 0 0
    children [
        Box {
          size 0.8 0.5 0.1  # Width, height, depth of the cuboid
        }
    ]
  }
      physics Physics {
  }
  }
}

